libraries/standard/http/src/http_client.c

```diff
@@ -930,18 +946,102 @@ HTTPStatus_t HTTPClient_Send( const HTTPTransportInterface_t * pTransport,
 /*-----------------------------------------------------------*/
 
 HTTPStatus_t HTTPClient_ReadHeader( const HTTPResponse_t * pResponse,
-                                    const char * pName,
-                                    size_t nameLen,
-                                    char ** pValue,
-                                    size_t * valueLen )
+                                    const uint8_t * pHeaderName,
+                                    size_t headerNameLen,
+                                    const uint8_t ** pHeaderValueLoc,
+                                    size_t * pHeaderValueLen )
 {
-    /* Disable unused parameter warnings. */
-    ( void ) pResponse;
-    ( void ) pName;
-    ( void ) nameLen;
-    ( void ) pValue;
-    ( void ) valueLen;
-    return HTTP_NOT_SUPPORTED;
+    HTTPStatus_t returnStatus = HTTP_SUCCESS;
+    HTTPParsingContext_t parsingContext;
+
+    memset( &parsingContext, 0, sizeof( HTTPParsingContext_t ) );
+
+    if( pResponse == NULL )
+    {
+        IotLogError( "Parameter check failed: pResponse is NULL." );
+        returnStatus = HTTP_INVALID_PARAMETER;
+    }
+    else if( pResponse->pBuffer == NULL )
+    {
+        IotLogError( "Parameter check failed: pRequestHeaders->pBuffer is NULL." );
+        returnStatus = HTTP_INVALID_PARAMETER;
+    }
+    else if( pResponse->bufferLen == 0u )
+    {
+        IotLogError( "Parameter check failed: pRequestHeaders->bufferLen is 0: Buffer len should be > 0." );
+        returnStatus = HTTP_INVALID_PARAMETER;
+    }
+    else if( pHeaderName == NULL )
+    {
+        IotLogError( "Parameter check failed: Input header name is NULL." );
+        returnStatus = HTTP_INVALID_PARAMETER;
+    }
+    else if( headerNameLen == 0u )
+    {
+        IotLogError( "Parameter check failed: Input header name length is 0: pHeaderName should be > 0." );
```

Comment:
`pHeaderName should be > 0.` --> `headerNameLen should be > 0.`

Reply:
Fixed